submarine-server/server-submitter/submitter-k8s/src/test/java/org/apache/submarine/server/submitter/k8s/K8SJobSubmitterTest.java

```diff
@@ -19,88 +19,188 @@
+import com.google.gson.GsonBuilder;
+import io.kubernetes.client.ApiException;
+import io.kubernetes.client.apis.CoreV1Api;
+import io.kubernetes.client.models.V1Namespace;
+import io.kubernetes.client.models.V1NamespaceList;
+import org.apache.submarine.server.api.exception.InvalidSpecException;
+import org.apache.submarine.server.api.job.Job;
+import org.apache.submarine.server.api.spec.JobSpec;
 import org.apache.submarine.server.submitter.k8s.model.CustomResourceJob;
 import org.apache.submarine.server.submitter.k8s.model.CustomResourceJobList;
+import org.apache.submarine.server.submitter.k8s.model.pytorchjob.PyTorchJob;
+import org.apache.submarine.server.submitter.k8s.model.tfjob.TFJob;
+import org.junit.After;
 import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Test;
 
 import java.io.File;
 import java.io.IOException;
+import java.io.Reader;
 import java.net.URISyntaxException;
 import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
 
+/**
+ * We have two ways to test submitter for K8s cluster, local and travis CI.
+ * <p>
+ * For running the tests locally, ensure that:
+ * 1. There's a k8s cluster running somewhere
+ * 2. There's a "config" file as ~/.kube/config targeting the cluster
+ * 3. A namespace submarine was created
+ * 3. The CRDs was created beforehand. The operator doesn't needs to be running.
+ * <p>
+ * Use "kubectl -n submarine get tfjob" or "kubectl -n submarine get pytorchjob"
+ * to check the status if you comment the deletion job code in method "after()"
+ * <p>
+ * <p>
+ * For the travis CI, we use the kind to setup K8s, more info see '.travis.yml' file.
+ * Local: docker run -it --privileged -p 8443:8443 -p 10080:10080 bsycorp/kind:latest-1.15
+ * Travis: See '.travis.yml'
+ */
 public class K8SJobSubmitterTest {
-  private final String jobName = "mnist";
+  private final String tfJobName = "mnist";
+  private final String pytorchJobName = "pytorch-dist-mnist-gloo";
+
+  // The spec files in test/resources
+  private final String tfJobSpecFile = "/tf_job_mnist.json";
+  private final String tfJobReqFile = "/tf_mnist_req.json";
+  private final String pytorchJobSpecFile = "/pytorch_job_mnist_gloo.json";
+  private final String pytorchJobReqFile = "/pytorch_job_req.json";
+
   private K8sJobSubmitter submitter;
-  private K8sJobRequest.Path path;
-
-  /**
-   * We have two ways to test submitter for K8s cluster, local and travis CI.
-   * For the travis CI, we use the kind to setup K8s, more info see '.travis.yml' file.
-   *  Local: docker run -it --privileged -p 8443:8443 -p 10080:10080 bsycorp/kind:latest-1.15
-   *  Travis: See '.travis.yml'
-   * @throws IOException IO
-   */
+
+  private K8sJobRequest.Path tfPath;
+  private K8sJobRequest.Path pyTorchPath;
+
   @Before
-  public void before() throws IOException {
+  public void before() throws IOException, ApiException {
     String confPath = System.getProperty("user.home") + "/.kube/config";
```

Comment:
I think the loading order for reading the k8s configuration file should be:
1. From the environment variable `KUBE_CONFIG`
2. From `~/.kube/config`
3. From `submarine-site.xml`

Reply:
For locally deployment of submarine server, the answer is yes. 
But for k8s depoyment, the submarine server should init the config from the cluster. I've code there but not tested. It could be tested using the submarine operator deployment way.
I'd like we can make the client initilization change a separate JIRA.